packages/medusa/src/repositories/money-amount.ts

```diff
@@ -103,13 +112,79 @@ export class MoneyAmountRepository extends Repository<MoneyAmount> {
       .execute()
   }
 
+  public async findManyForVariantInRegion(
+    variant_id: string,
+    region_id?: string,
+    currency_code?: string,
+    customer_id?: string,
+    includeDiscountPrices?: boolean
+  ): Promise<MoneyAmount[]> {
+    const date = new Date()
+
+    let qb = this.createQueryBuilder("ma")
+      .leftJoinAndSelect(
+        "ma.price_list",
+        "price_list",
+        "ma.price_list_id = price_list.id "
+      )
+      .where("ma.variant_id = :variant_id", { variant_id: variant_id })
+      .andWhere("(ma.price_list_id is null or price_list.status = 'active')")
+      .andWhere(
+        "(price_list is null or price_list.ends_at is null OR price_list.ends_at > :date) ",
+        {
+          date: date.toUTCString(),
+        }
+      )
+      .andWhere(
+        "(price_list is null or price_list.starts_at is null OR price_list.starts_at < :date)",
+        {
+          date: date.toUTCString(),
+        }
+      )
+
+    if (region_id || currency_code) {
+      qb = qb.andWhere(
+        "(ma.region_id = :region_id OR ma.currency_code = :currency_code)",
+        {
+          region_id: region_id,
+          currency_code: currency_code,
+        }
+      )
+    } else if (!customer_id && !includeDiscountPrices) {
+      qb = qb.andWhere("price_list is null")
+    }
+
+    if (customer_id) {
+      qb = qb
+        .leftJoinAndSelect("price_list.customer_groups", "cgroup")
+        .leftJoinAndSelect(
+          "customer_group_customers",
+          "cgc",
+          "cgc.customer_group_id = cgroup.id"
+        )
+        .andWhere("(cgc is null OR cgc.customer_id = :customer_id)", {
+          customer_id,
+        })
```

Comment:
`innerJoin(....., "cgc.customer_group_id = cgroup.id AND cgc.customer_id = :customer_id")`